scripts/Diffraction/isis_powder/test/ISISPowderPearlTest.py

```diff
@@ -0,0 +1,39 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from unittest.mock import patch
+import mantid.simpleapi as mantid   # noqa: F401 # runs .pth file to add scripts subdirectories
+from isis_powder.pearl import Pearl
+from isis_powder.pearl_routines import pearl_advanced_config
+
+
+class ISISPowderPearlTest(unittest.TestCase):
+
+    @patch("isis_powder.routines.focus.focus")
+    def test_long_mode_on(self, mock_focus):
+        def check_long_mode_on_inst(*args, **kwargs):
+            inst = kwargs.get("instrument")
+            self.assertEqual(inst._inst_settings.long_mode, True)
+            self.assertEqual(inst._inst_settings.raw_data_crop_vals,
+                             pearl_advanced_config.long_mode_on_params["raw_data_tof_cropping"])
+
+        mock_focus.side_effect = check_long_mode_on_inst
+        # firstly set long_mode as default in the Pearl object
+        inst_obj = Pearl(user_name="PEARL", calibration_directory="dummy", output_directory="dummy",
+                         do_absorb_corrections=False, perform_attenuation=False, vanadium_normalisation=False, long_mode=True)
+        inst_obj.focus(run_number=999)
+        mock_focus.assert_called_once()
+
+        # now activate it in the call to focus
+        inst_obj = Pearl(user_name="PEARL", calibration_directory="dummy", output_directory="dummy",
+                         do_absorb_corrections=False, perform_attenuation=False, vanadium_normalisation=False, long_mode=False)
+        inst_obj.focus(run_number=999, long_mode=True)
```

Comment:
Almost. 1) is correct. for 2) if the instrument long_mode is False and the call to focus has long_mode=True the parameter value in the call to Focus overrides the one in the constructor of the Pearl object and the focus is only done in long mode

Reply:
Yes it might be useful to check instrument having long_mode=True followed by focus long_mode=False

Reply:
> Almost. 1) is correct. for 2) if the instrument long_mode is False and the call to focus has long_mode=True the parameter value in the call to Focus overrides the one in the constructor of the Pearl object and the focus is only done in long mode

Ah OK that makes sense, you assert
```
self.assertEqual(mock_focus.call_count, 2)
```
one of those calls is from the above test where you don't specify long_mode in the focus. It might be better to reset_mock?

Reply:
Yes that's where the "2" has come from and yes would be clearer to reset in between and check the call count is "1" each time